File: benchmarks/correctness/real_world/python/reverse_complement.py
import json
from Bio import SeqIO
from Bio.Seq import Seq

results = []
for i, record in enumerate(SeqIO.parse("real_data/yeast_genome.fa", "fasta")):
    if i >= 5:
        break
    sub = str(record.seq)[:200].upper()
    rc = str(Seq(sub).reverse_complement())
    results.append({"id": record.id, "original": sub, "revcomp": rc})

print(json.dumps({"sequences": results}))

File: benchmarks/correctness/real_world/r/reverse_complement.R
library(Biostrings)
library(jsonlite)
seqs <- readDNAStringSet("real_data/yeast_genome.fa")
results <- list()
for (i in 1:min(5, length(seqs))) {
  sub_seq <- subseq(seqs[[i]], 1, min(200, width(seqs[[i]])))
  original <- as.character(sub_seq)
  rc <- as.character(reverseComplement(sub_seq))
  id <- sub("\\s.*", "", names(seqs)[i])
  results[[i]] <- list(id = id, original = toupper(original), revcomp = toupper(rc))
}
cat(toJSON(list(sequences = results), auto_unbox = TRUE))
cat("\n")
